dali/python/nvidia/dali/pipeline.py

```diff
@@ -77,14 +75,12 @@ class Pipeline(object):
         Negative values for this parameter are invalid - the default
         value may only be used with serialized pipeline (the value
         stored in serialized pipeline is used instead).
-    device_id : int, optional, default = -1
+    device_id : int, optional, default = None
         Id of GPU used by the pipeline.
-        A None value for this parameter means that DALI should not use GPU nor CUDA runtime.
-        This limits the pipeline to only CPU operators but allows it to run on any CPU capable
-        machine.
-    seed : int, optional, default = -1
-        Seed used for random number generation. Leaving the default value
-        for this parameter results in random seed.
+        If the pipeline requires a GPU and this field is left unspecified, DALI will use the
+        current CUDA device (according to ``cudaGetDevice``) at the time the pipeline is built.
```

Comment:
With this change, how should user express CPU-only pipeline?

Reply:
They shouldn't :) When None is passed, the GPU won't be used unless its necessary for the pipeline to run. I see little to no utility in forcing a pipeline to reject GPU operators altogether.

Reply:
For me it can go either way but based on my experience with benchmarking it might be useful to be able to quickly tell in the code that I want this pipeline to be on the CPU. It's creation code might be very convoluted and spread out, device propagation through it imperfect. When I say in the constructor that I want the pipeline on the CPU and get a runtime error I now that something went wrong. Otherwise, if I miss some kind of operator using GPU somewhere I get GPU based pipeline without realizing it. But this is the only case that comes to my mind. As I said for me it can go either way.

Reply:
That way we could go back to explicitly passing `CPU_ONLY_DEVICE_ID`.